essentials-framework/Essentials Core/PepperDashEssentialsBase/Devices/GenericIRController.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using Crestron.SimplSharpPro.DeviceSupport;
+using Newtonsoft.Json;
+using PepperDash.Core;
+using PepperDash.Essentials.Core;
+using PepperDash.Essentials.Core.Bridges;
+using PepperDash.Essentials.Core.Config;
+
+namespace PepperDash_Essentials_Core.Devices
+{
+    public class GenericIrController: EssentialsBridgeableDevice
+    {
+        //data storage for bridging
+        private BasicTriList _trilist;
+        private uint _joinStart;
+        private string _joinMapKey;
+        private EiscApiAdvanced _bridge;
+
+        private readonly IrOutputPortController _port; 
+
+        public string[] IrCommands {get { return _port.IrFileCommands; }}
+
+        public GenericIrController(string key, string name, IrOutputPortController irPort) : base(key, name)
+        {
+            _port = irPort;
+
+            if (_port == null)
+            {
+                Debug.Console(0, this, Debug.ErrorLogLevel.Error, "IR Port is null, device will not function");
+                return;
+            }
+            DeviceManager.AddDevice(_port);
```

Comment:
Need to double check that the IRPortController doesn't already add itself to the DeviceManager

Reply:
Just checked, and it does not. I'll make an issue to change that, as it affects more than just this class.